Bottom: a1a9cfa9f0b3da7e4c7b4eac32d592f3195c8afa
Top:    7c2948c4f3c3f4451c9ff850fbd8c7736afc91f4
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-11-15 19:42:27 +0000

hid/gtk: Add a guard against GC's being passed from another HID

Various other HIDs have this check.


---

diff --git a/src/hid/gtk/gtkhid-gdk.c b/src/hid/gtk/gtkhid-gdk.c
index 51a6f22..dc2ad98 100644
--- a/src/hid/gtk/gtkhid-gdk.c
+++ b/src/hid/gtk/gtkhid-gdk.c
@@ -441,6 +441,13 @@ ghid_set_line_cap_angle (hidGC gc, int x1, int y1, int x2, int y2)
 static int
 use_gc (hidGC gc)
 {
+
+  if (gc->me_pointer != &ghid_hid)
+    {
+      fprintf (stderr, "Fatal: GC from another HID passed to GTK HID\n");
+      abort ();
+    }
+
   if (!gport->pixmap)
     return 0;
   if (!gc->gc)
